### DISCUSSION TITLE
Strict enum attribute

### DISCUSSION THREAD

**Discussion by TahirAhmadov** [dotnet/runtime#61966] [Ideas]
Can we get an attribute which can be applied to enums which makes them "strict"?
```cs
enum Foo { A = 1, B = 2, C = 3, }
var foo = (Foo)1; // OK
var foo2 = (Foo)4; // OK

[StrictEnum]
enum Bar { X = 1, Y = 2, Z = 3, }
var bar = (Bar)1; // OK
var bar2 = (Bar)4; // runtime error

class Class
{
  Bar bar; // compilation error: valid value has to be assigned (this would be a C# change)
  Bar bar2 = default; // should this be same as previous line, compilation error? or runtime exception? or nothing, since it's just a zero-memset behind the scenes?
}
```

**Comment by huoyaoyuan** [comment]
See https://github.com/dotnet/csharplang/issues/3179

**Reply by TahirAhmadov** [reply]
Thanks for pointing me in that direction. I asked a couple of question on that issue. It does seem to be a much more involved improvement, which is tied in with even more involved improvements. I'm curious if people would be interested in just simply adding an attribute to improve existing enums, before we get around to implementing closed type hierarchies and all that.
